mantidproject/mantid
# A few changes to support msvc 2017

Fixes warnings coming from the MSVC 2017 build.

**To test:**

Build locally using msvc 2017 to ensure everything builds cleanly and they are no build or warnings. Also ensure the 2015 build still works,

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
